R/as_draws_df.R

```diff
@@ -171,22 +163,22 @@ as_draws_df.mcmc.list <- function(x, ...) {
   }
   if (has_chain_column) {
     chain_ids <- x[[.chain]]
-    x[[.chain]] <- NULL
+    x[.chain] <- NULL
   } else {
-    chain_ids <- rep(1L, NROW(x))
+    chain_ids <- rep(1L, nrow(x))
   }
   # prepare draw indices --- i.e. drop them, since they are regenerated below
-  x[[".draw"]] <- NULL
+  x[".draw"] <- NULL
 
   # add reserved variables to the data
   check_new_variables(names(x))
-  x$.chain <- chain_ids
-  x$.iteration <- iteration_ids
+  x[".chain"] <- chain_ids
```

Comment:
Why these changes from `$` to `[ ]`? As far as I can tell, these are always equivalent in the assignment case and also equivalent in the extraction case if we know the columns to exist in the df?

Reply:
Yeah, they are equivalent. For some reason, I always thought using brackets + name string was better practice in package/shiny development but cannot find a reference at the moment ([other than this old article](https://www.r-bloggers.com/2009/10/r-accessors-explained/)) to support it.

Would you like me to revert these?

Reply:
Thanks! It is fine for me to leave it like this, just wanted to understand if there was a reason I was not aware of.

Reply:
> I always thought using brackets + name string was better practice in package/shiny development

I think you're right because using `$` allows for partial matching, which is a bit dangerous. In this case I don't think there's a chance of partial matching so it doesn't matter, but in general I think this is why `[` is preferred to `$`.